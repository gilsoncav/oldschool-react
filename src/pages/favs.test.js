import Favs from './favs';

describe('Favs page component', () => {
  it('renders correctly when loading', () => {
    const wrapper = shallow(
      <Favs.WrappedComponent
        isLoading
        favList={[]}
        requestFavList={() => {}}
        searchFavList={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when not loading and has games', () => {
    const wrapper = shallow(
      <Favs.WrappedComponent
        isLoading={false}
        favList={[{
          VideoURL: undefined, Name: 'Game', Platform: 'Game', Overview: 'Overview',
        }]}
        requestFavList={() => {}}
        searchFavList={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when not loading and not has games', () => {
    const wrapper = shallow(
      <Favs.WrappedComponent
        isLoading={false}
        favList={[]}
        requestFavList={() => {}}
        searchFavList={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

