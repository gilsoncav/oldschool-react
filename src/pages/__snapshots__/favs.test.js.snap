// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favs page component renders correctly when loading 1`] = `
<LayoutAuth>
  <div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <Row
      gutter={0}
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <Col
        offset={6}
        span={12}
      >
        <Input
          className="game-search"
          onChange={[Function]}
          onPressEnter={[Function]}
          placeholder="Search Fav Games"
          prefix={
            <Icon
              style={
                Object {
                  "color": "#b3b3b3",
                }
              }
              type="search"
            />
          }
          size="large"
          style={
            Object {
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </Col>
    </Row>
    <Row
      gutter={24}
      style={
        Object {
          "padding": "30px 15px",
        }
      }
    >
      <GameList
        gameOnClick={[Function]}
        gamesArray={Array []}
        isLoading={true}
        maxTotalGames={24}
      />
    </Row>
  </div>
</LayoutAuth>
`;

exports[`Favs page component renders correctly when not loading and has games 1`] = `
<LayoutAuth>
  <div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <Row
      gutter={0}
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <Col
        offset={6}
        span={12}
      >
        <Input
          className="game-search"
          onChange={[Function]}
          onPressEnter={[Function]}
          placeholder="Search Fav Games"
          prefix={
            <Icon
              style={
                Object {
                  "color": "#b3b3b3",
                }
              }
              type="search"
            />
          }
          size="large"
          style={
            Object {
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </Col>
    </Row>
    <Row
      gutter={24}
      style={
        Object {
          "padding": "30px 15px",
        }
      }
    >
      <GameList
        gameOnClick={[Function]}
        gamesArray={
          Array [
            Object {
              "Name": "Game",
              "Overview": "Overview",
              "Platform": "Game",
              "VideoURL": undefined,
            },
          ]
        }
        isLoading={false}
        maxTotalGames={24}
      />
    </Row>
  </div>
</LayoutAuth>
`;

exports[`Favs page component renders correctly when not loading and not has games 1`] = `
<LayoutAuth>
  <div
    style={
      Object {
        "height": "100vh",
        "overflow": "auto",
      }
    }
  >
    <Row
      gutter={0}
      style={
        Object {
          "marginTop": "2vh",
        }
      }
    >
      <Col
        offset={6}
        span={12}
      >
        <Input
          className="game-search"
          onChange={[Function]}
          onPressEnter={[Function]}
          placeholder="Search Fav Games"
          prefix={
            <Icon
              style={
                Object {
                  "color": "#b3b3b3",
                }
              }
              type="search"
            />
          }
          size="large"
          style={
            Object {
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </Col>
    </Row>
    <Row
      gutter={24}
      style={
        Object {
          "padding": "30px 15px",
        }
      }
    >
      <GameList
        gameOnClick={[Function]}
        gamesArray={Array []}
        isLoading={false}
        maxTotalGames={24}
      />
    </Row>
  </div>
</LayoutAuth>
`;
